### DISCUSSION TITLE
CSLA 8 compile in the solution.

### DISCUSSION THREAD

**Discussion by HakkeunLee** [MarimerLLC/csla#3944] [Questions]
Hello All,

I have usually included CSLA source project directly into solution, not using nuget package, to get more detail info on debug and
to learn how CSLA works internally.
On upgrading Blazor WASM based on CSLA 7 to CSLA 8.x, which I only changed CSLA project source from v7 to v8.x,
it emits numerous errors on compile as if there were no referenced base libs such as 'using system',etc.
When compiled CSLA alone in seperate solution, it did successfully.
I encountered this situation on CSLA 7 as well, but it compiled fine in the solution when I set the target framework 'net7.0' only,
but not in CSLA 8.x, though I did same as CSLA 7.

I think I missed something simple but can't get it even struggling with comparing the proj file of both. :(
What am I missing ?

Thanks in advance.

HK.Lee

**Comment by rockfordlhotka** [comment]
Are you able to clone the Csla repo, open the `csla.build.sln`, and build that solution?

If that doesn't work, it is probably because you need to install missing workloads for Visual Studio or .NET.

**Comment by HakkeunLee** [comment]
I appreciate your direct answer for my stupid question.
It builds fine with csla.build.sln and even csla project itself, 
but failed in my solution when I add csla project and reference it with my business project.

Why I asked is that I had the error as follows when I run the csla 8.1.1 package
and fetch auth info via CredentialValidator from the server, 
and I added clsa souce project in the solution so that I can get what is going on inside of Csla.

```
Message = 'Object reference not set to an instance of an object.'
SerializationStackTraceString = '   at Csla.Blazor.WebAssembly.ApplicationContextManager.GetClientContext(ExecutionLocations executionLocation)\n   at Csla.ApplicationContext.get_ClientContext()\n   at Csla.ApplicationContext.get_RuleSet()\n   at Csla.Rules.BusinessRules.HasPermission(App...
```

Csla config is same as the client's config of project tracker sample.
Though I set  'SyncContextWithServer = false' , the error is same with 'SyncContextWithServer = true'.

```c#
builder.Services.AddCsla(o => o
  .AddBlazorWebAssembly(o => o.SyncContextWithServer = false)
  .DataPortal(o => o.ClientSideDataPortal(o => o
    .UseHttpProxy(o => o.DataPortalUrl = "/api/dataportal"))));
```

I have read couple of times of your blog about state manager,etc, but I can't grasp completely what i should do in real senario.
It looks I need more time to learn and understand, and I decided to keep Csla 7.0/Blazor 7.0 while I  did.

Thanks.
HK.Lee

**Reply by rockfordlhotka** [reply]
If you are manually referencing the CSLA projects, then you need to make sure you reference all of the required projects for each of _your_ project types.

For example, a Blazor server (aspnetcore) project must reference

* Csla
* Csla.AspNetCore
* Csla.Blazor

A Blazor client (wasm) project must reference:

* Csla
* Csla.Blazor
* Csla.Blazor.WebAssembly

A business library needs to reference

* Csla

A Blazor server project needs to be configured to use aspnetcore and blazor:

```c#
builder.Services.AddCsla(o => o
  .AddAspNetCore()
  .AddBlazorWebAssembly(o => o.SyncContextWithServer = false)
  .DataPortal(o => o.ClientSideDataPortal(o => o
    .UseHttpProxy(o => o.DataPortalUrl = "/api/dataportal"))));
```

Also, to use the Blazor 8 state management, your server project must implement a `CslaStateController`, which is what the wasm client uses to get/put the session state, and what it uses to get the current user identity from the server.

**Comment by HakkeunLee** [comment]
Thank you, Rocky.

I will invetigate more time to adapt Csla 8 and try to make a sample project on git for WASM config.

Thanks.
HK.Lee
